### PR TITLE
[new] Add monaco-editor@0.7.0 w/ npm auto-update

### PR DESCRIPTION
**Pull request by alexandrudima**
Reference issue #9351
# Profile of the lib
- Git repository (required): https://github.com/Microsoft/monaco-editor
- Official website (optional, not the repository): https://microsoft.github.io/monaco-editor/
- NPM package url (optional): https://www.npmjs.com/package/monaco-editor
- GitHub / Bitbucket popularity (required):
  - Count of stars: 3,212
  - Count of watchers: 152
  - Count of forks: 178
- NPM download stats (optional):
  - Downloads in the last day: 243
  - Downloads in the last week: 2,960
  - Downloads in the last month: 10,636
# Essential checklist
- [x] I'm the author of this library
  - [ ] **Sorry, not sure what this means** I would like to add link to the page of this library on CDNJS on website / readme
- [x] This lib was not found on cdnjs repo
- [x] No already exist / duplicated issue and PR
- [x] The lib has notable popularity
  - [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  - [x] More than 500 downloads stats per month on npm registry
- [x] Project has public repository on famous online hosting platform (or been hosted on npm)
# Auto-update checklist
- [x] **uses npm** Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [ ] **?** Auto-update target/source is valid.
- [x] Auto-update filemap is correct.
# Git commit checklist
- [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
- [x] The parent of the commit(s) in the PR is not old than 3 days.
- [ ] **sorry, I used the master branch in my fork... I can change that if it is really necessary** Pull request is sending from a non-master branch with meaningful name.
- [x] Separate unrelated changes into different commits.
- [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
- [x] Close corresponding issue in commit message
- [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by kennynaoh** [comment]
Hi @alexandrudima ,
I think the main file is `editor.main.js`, right?
Please add `editor.main.js` as `filename` field in `package.json`.
And please also add `https://microsoft.github.io/monaco-editor/` as `homepage` field and `keywords` field in `package.json`
Thanks for your help!

**Comment by kennynaoh** [comment]
Please remove these fields, `private`, `scripts`, `typings`, `devDependencies` in `package.json`.
Thanks!

File: ajax/libs/monaco-editor/package.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "monaco-json": "1.1.1",
+    "monaco-languages": "0.6.0",
+    "monaco-typescript": "2.0.1",
+    "rimraf": "^2.5.2",
+    "typedoc": "^0.5.0",
+    "uncss": "^0.14.1"
+  },
+  "npmName": "monaco-editor",
+  "npmFileMap": [
+    {
```
Comment (kennynaoh) [inline ajax/libs/monaco-editor/package.json]: Please add `basePath` field.
Thanks!

**Comment by x09326** [comment]
@alexanderdickson 
You can you the **fixformat.js** to modify the **package.json** in the **tools** folder.

**Comment by alexandrudima** [comment]
@kennynaoh @x09326 Thank you for the review, I have updated `package.json` as you described and recreated the commit.

File: ajax/libs/monaco-editor/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "TypeScript",
+    "JavaScript"
+  ],
+  "npmName": "monaco-editor",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "min/**/*",
+        "min-maps/**/*"
```
Comment (kennynaoh) [inline ajax/libs/monaco-editor/package.json]: How do you think to use this pattern?
`+(min|min-maps)/**/*`

**Comment by alexandrudima** [comment]
@kennynaoh Yes, that would make sense. I'm sorry I cannot do that change now as I have this repository checked out on another machine.

**Comment by kennynaoh** [comment]
cc @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@alexandrudima just edit online is fine, I'll squash the commits into one so the commit history will still be good.

@kennynaoh what's the problem here? I didn't see the reason you make it pending for 6 days and then tag me.

**Comment by PeterDaveHello** [comment]
@x09326 @nolsherry would you guys please also help review this?

**Comment by kennynaoh** [comment]
@PeterDaveHello 
I suggest author to change `files` field, but he can't do it recently.
That's the reason.
The other parts I think is good to merge.
If there aren't any other problems, could you help change the `files` field to `+(min|min-maps)/**/*` in the merge commit?
Thanks!

**Comment by alexandrudima** [comment]
Sorry about it, I was overloaded with VS Code issues :). Here is the change, I did it in the web ui.

File: ajax/libs/monaco-editor/package.json [hunk 3]
```diff
@@ -21,8 +21,7 @@
     {
       "basePath": "",
       "files": [
-        "min/**/*",
-        "min-maps/**/*"
+        "(min|min-maps)/**/*"
```
Comment (PeterDaveHello) [inline ajax/libs/monaco-editor/package.json]: Should be `+"(min|min-maps)/**/*"`

**Comment by PeterDaveHello** [comment]
@alexandrudima any updates here?

File: ajax/libs/monaco-editor/package.json (hunk 3 above)
Comment (joaomoreno) [inline ajax/libs/monaco-editor/package.json]: Why the `+`?

Comment (PeterDaveHello) [inline ajax/libs/monaco-editor/package.json]: https://github.com/isaacs/node-glob#glob-primer

Comment (joaomoreno) [inline ajax/libs/monaco-editor/package.json]: Do you mind explaining to me what the difference between `(min|min-maps)/**/*` and `+(min|min-maps)/**/*` is?

Meanwhile, I'll ping @alexandrudima so we get to wrap this up.

Comment (joaomoreno) [inline ajax/libs/monaco-editor/package.json]: Ah! Had no clue. 😮

**Comment by alexandrudima** [comment]
:+1: I'll give @joaomoreno write access to my fork, he will pick it up from here.

**Comment by joaomoreno** [comment]
Pushed.

File: ajax/libs/monaco-editor/package.json (hunk 3 above)
Comment (PeterDaveHello) [inline ajax/libs/monaco-editor/package.json]: Nope, please take a look at https://github.com/isaacs/node-glob#glob-primer

**Comment by PeterDaveHello** [comment]
ping @kennynaoh

File: ajax/libs/monaco-editor/package.json (hunk 3 above)
Comment (joaomoreno) [inline ajax/libs/monaco-editor/package.json]: Here's `+(min|min-maps)/**/*`:

```
→ node -p 'require("minimatch")("min/file.js", "+(min|min-maps)/**/*")'
true
→ node -p 'require("minimatch")("minmin/file.js", "+(min|min-maps)/**/*")'
true
→ node -p 'require("minimatch")("minmin-maps/file.js", "+(min|min-maps)/**/*")'
true
```

Here's `{min,min-maps}/**/*`:

```
→ node -p 'require("minimatch")("min/file.js", "{min,min-maps}/**/*")'
true
→ node -p 'require("minimatch")("minmin/file.js", "{min,min-maps}/**/*")'
false
→ node -p 'require("minimatch")("minmin-maps/file.js", "{min,min-maps}/**/*")'
false
```

I've changed it to `{min,min-maps}/**/*`.

Comment (PeterDaveHello) [inline ajax/libs/monaco-editor/package.json]: We are not using `{`, `}` here, should be `@(min|min-maps)/**/*`

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by joaomoreno** [comment]
We are waiting for @kennynaoh to reply.

**Comment by joaomoreno** [comment]
Jesus Christ guys. We simply give up. Please close this.

**Comment by PeterDaveHello** [comment]
@joaomoreno sorry for that, I think there is a problem with @kennynaoh , @pvnr0082t can you take this?

**Comment by alexandrudima** [comment]
I will close this now. All the best!

**Pull request closed without merging**